### PR TITLE
Keep Sidebar Visible

### PR DESCRIPTION
**Pull request by demonpig**
This commit is simply adding a switch on the PreferenceDialog to prevent the sidebar from disappearing when in EDIT mode. When the switch is off, the sidebar will disappear when in EDIT mode. The default value for the switch is "off."

screenshot from 2017-03-06 21-51-12

### CODE REVIEW

File: src/Widgets/PreferencesDialog.vala [hunk 1]
```diff
@@ -19,6 +19,45 @@
 * Authored by: Felipe Escoto <felescoto95@hotmail.com>
 */
 
+/**
+ *  The purpose of this class is to easily create a new Switchbox for the
+ *  PreferencesDialog class.
+ */
+private class ENotes.PreferencesDialogSwitchBox {
+
```
Comment (Philip-Scott) [inline src/Widgets/PreferencesDialog.vala]: Private classes should be inside and at the bottom of the class you're going to use it at :wink:

Comment (Philip-Scott) [inline src/Widgets/PreferencesDialog.vala]: But you know... I like the idea. But i would remove it at least for this PR. If you still want, you can make another PR that changes all labes + switches to using this :) 

_Small diff is best diff_

File: src/Widgets/PreferencesDialog.vala [hunk 2]
```diff
@@ -31,6 +70,11 @@ public class ENotes.PreferencesDialog : Gtk.Dialog {
     private Gtk.Switch indent_switch;
     private Gtk.Switch line_numbers_switch;
 
+    /**
+     *  Toggles the setting to keep sidebar (Notebooks) visible
+     */
```
Comment (Philip-Scott) [inline src/Widgets/PreferencesDialog.vala]: No need for the comments here :wink:

File: src/Widgets/PreferencesDialog.vala [hunk 3]
```diff
@@ -19,6 +19,45 @@
...
+ *  PreferencesDialog class.
+ */
+private class ENotes.PreferencesDialogSwitchBox {
+
+    private Gtk.Box container;
+    private Gtk.Switch switchbox;
+    private Gtk.Label  label;
+
+    public Gtk.Box SwitchBox { get { return this.container; } }
+    public Gtk.Label SwitchBoxLabel { get { return this.label; } }
```
Comment (Philip-Scott) [inline src/Widgets/PreferencesDialog.vala]: variable_names should be cased_like_this 

https://elementary.io/docs/code/reference#variable-names-class-names-function-names

File: src/Widgets/ViewEditStack.vala [hunk 4]
```diff
@@ -100,7 +100,10 @@ public class ENotes.ViewEditStack : Gtk.Overlay {
         current_mode = ENotes.Mode.EDIT;
         Headerbar.get_instance ().set_mode (ENotes.Mode.EDIT);
         stack.set_visible_child_name ("editor");
-        Sidebar.get_instance().visible = false;
+        if ( ! settings.keep_sidebar_visible)
```
Comment (Philip-Scott) [inline src/Widgets/ViewEditStack.vala]: should be:

```
if (!settings.keep_sidebar_visible) {

}
```

File: src/Widgets/PreferencesDialog.vala [hunk 5]
```diff
@@ -119,6 +163,8 @@ public class ENotes.PreferencesDialog : Gtk.Dialog {
         var switch_box_ln = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 0);
         switch_box_ln.add (line_numbers_switch);
 
+        sidebar_switch = new ENotes.PreferencesDialogSwitchBox ("Keep Sidebar Visible:", settings.keep_sidebar_visible);
```
Comment (Philip-Scott) [inline src/Widgets/PreferencesDialog.vala]: When using translations, the translated marks need to be on the string itself, not like on line 44 on PreferencesSwitchBox

https://elementary.io/docs/code/getting-started#Adding-Translations

File: src/Widgets/PagesList.vala [hunk 6]
```diff
@@ -279,8 +279,7 @@ public class ENotes.PagesList : Gtk.Box {
 
         listbox.row_activated.connect ((row) => {
             window.toggle_edit ();
-
-            if (ENotes.ViewEditStack.current_mode == Mode.EDIT) {
+            if (ENotes.ViewEditStack.current_mode == Mode.EDIT && settings.keep_sidebar_visible) {
                 ENotes.Editor.get_instance ().give_focus ();
```
Comment (Philip-Scott) [inline src/Widgets/PagesList.vala]: Editor should always gain focus if you move to the editor. You're moving to the editor for a reason aren't you? ;)

File: src/Services/Settings.vala [hunk 7]
```diff
@@ -35,6 +35,11 @@ public class ENotes.Services.Settings : Granite.Services.Settings {
     public bool line_numbers {get; set;}
     public bool import_files {get; set;}
 
+    /**
+     *  Keep sidebar visible when switching between VIEW and EDIT modes
+     */
```
Comment (Philip-Scott) [inline src/Services/Settings.vala]: When you comment, you should use the compressed whenever you can

https://elementary.io/docs/code/reference#comments

File: src/Widgets/ViewEditStack.vala (hunk 4 above)
Comment (Philip-Scott) [inline src/Widgets/ViewEditStack.vala]: You could even go a step further and: ` Sidebar.get_instance().visible = !settings.keep_sidebar_visible`

File: src/Widgets/PreferencesDialog.vala [hunk 8]
```diff
@@ -19,6 +19,45 @@
...
+ *  The purpose of this class is to easily create a new Switchbox for the
+ *  PreferencesDialog class.
+ */
+private class ENotes.PreferencesDialogSwitchBox {
+
+    private Gtk.Box container;
+    private Gtk.Switch switchbox;
+    private Gtk.Label  label;
+
+    public Gtk.Box SwitchBox { get { return this.container; } }
```
Comment (Philip-Scott) [inline src/Widgets/PreferencesDialog.vala]: You don't need to separate both the public and private variables like you did here :) (At least not in this case since it's a simple return)

But you can do:

`public Gtk.Box switch_box { get; private set; }`

File: src/Widgets/PreferencesDialog.vala (hunk 5 above)
Comment (demonpig) [inline src/Widgets/PreferencesDialog.vala]: Thanks did not know that. Will get it corrected.

File: src/Services/Settings.vala (hunk 7 above)
Comment (demonpig) [inline src/Services/Settings.vala]: Whoops! A lot of comments will carry over from C# and PHP programming. Will follow that.

File: src/Widgets/PreferencesDialog.vala (hunk 8 above)
Comment (demonpig) [inline src/Widgets/PreferencesDialog.vala]: Good call. Will get that corrected.

File: src/Widgets/PreferencesDialog.vala (hunk 1 above)
Comment (demonpig) [inline src/Widgets/PreferencesDialog.vala]: Are you cool with me just keeping the class? I will add it to the PreferencesDialog class. I do not think will harm anything.

Comment (Philip-Scott) [inline src/Widgets/PreferencesDialog.vala]: For the Commit history, i'd rather it be a different commit :smile: (It's one of the things we do at the rest of the elementary projects. It's better to have many small commits than large commits, specially with PRs)

Comment (demonpig) [inline src/Widgets/PreferencesDialog.vala]: Alright. I will remove it then.

Comment (Philip-Scott) [inline src/Widgets/PreferencesDialog.vala]: Thank you sir :)
